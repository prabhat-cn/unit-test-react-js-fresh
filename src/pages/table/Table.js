import React from 'react';

const Table = () => {
  return (
    <center>
      <div className="tableCls custom-border" data-test="component-table">
        <table>
          <thead>
            <tr>
              <th>#Sl.No</th>
              <th>Employee Name</th>
              <th>Company Name</th>
              <th>Address</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>1</td>
              <td>Rahul Kumar</td>
              <td>TCS</td>
              <td>America</td>
            </tr>
          </tbody>
        </table>
      </div>
    </center>
  );
};

export default Table;
