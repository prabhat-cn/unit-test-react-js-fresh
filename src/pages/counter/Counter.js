import React, { useState, Fragment } from 'react';
import Table from '../table/Table';

export const Counter = () => {
  const [count, setCount] = useState(0);
  const [error, setError] = useState(false);

  const decrement = () => {
    if (count <= 0) {
      setError(true);
      setTimeout(() => {
        setError(false);
      }, 1000);
      setCount(0);
    } else {
      setCount(count - 1);
    }
  };
  return (
    <Fragment>
      <div data-test="component-counter" className="counter-app">
        <h1 data-test="title-header">Click Counter</h1>
        <h1 data-test="counter-display">
          Counted Value &nbsp;{' '}
          <span data-test="count" className={`click-count-${count}`}>
            {count}
          </span>
        </h1>
        <p>
          {error && (
            <span style={{ color: 'yellow' }}>
              The value can not decrement than 0
            </span>
          )}
        </p>
        <button
          data-test="increment-button"
          onClick={() => {
            setError(false);
            setCount(count + 1);
          }}
        >
          Increment Counter
        </button>
        &nbsp; &nbsp; &nbsp;
        <button data-test="decrement-button" onClick={decrement}>
          Decrement Counter
        </button>
      </div>
      <br />
      <h3>Table Value To Check 'dive()'</h3>
      <Table />
    </Fragment>
  );
};
