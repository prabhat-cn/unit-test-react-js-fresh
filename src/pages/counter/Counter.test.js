import { shallow } from 'enzyme';
import '../../setupTests';
import Table from '../table/Table';
import { Counter } from './Counter';

describe('Counter Component', () => {
  // it('render Without any error is Present', () => {
  //   let wrapper = shallow(<Counter />);
  //   console.log(wrapper.debug());
  //   let isPresent = wrapper
  //     // .find('fragment')
  //     // .dive()
  //     .is('[data-test="component-counter"]');
  //   expect(isPresent).toBe(true);
  // });
  it('render Without any error & find length', () => {
    let wrapper = shallow(<Counter />);
    let isPresent = wrapper.find('[data-test="component-counter"]');
    expect(isPresent.length).toBe(1);
  });
  it('Click Count Increment', () => {
    let wrapper = shallow(<Counter />);
    // without click zero state by class
    let countZero = wrapper.find('.click-count-0');
    expect(countZero.length).toBe(1);
    // Count start 1 by simulate click
    wrapper.find('[data-test="increment-button"]').simulate('click');
    let countOne = wrapper.find('.click-count-1');
    expect(countOne.length).toBe(1);

    // or nother type
    let incrementCount = wrapper.find('[data-test="count"]').text();
    expect(incrementCount).toBe('1');
  });

  it('Click Count Increment 2', () => {
    let wrapper = shallow(<Counter />);
    // without click zero state by class
    let countZero = wrapper.find('.click-count-0');
    expect(countZero.length).toBe(1);
    // Count by simulate double click
    wrapper.find('[data-test="increment-button"]').simulate('click');
    wrapper.find('[data-test="increment-button"]').simulate('click');
    let countOne = wrapper.find('.click-count-2');
    expect(countOne.length).toBe(1);
  });

  it('Click Count Increment 3', () => {
    let wrapper = shallow(<Counter />);
    // without click zero state by class
    let countZero = wrapper.find('.click-count-0');
    expect(countZero.length).toBe(1);
    // Count by simulate triple click
    wrapper.find('[data-test="increment-button"]').simulate('click');
    wrapper.find('[data-test="increment-button"]').simulate('click');
    wrapper.find('[data-test="increment-button"]').simulate('click');
    let countOne = wrapper.find('.click-count-3');
    expect(countOne.length).toBe(1);
  });

  it('Click Count Decrement', () => {
    const wrapper = shallow(<Counter />);
    // Find the button
    wrapper.find('[data-test="decrement-button"]').simulate('click');
    let decrementCount = wrapper.find('[data-test="count"]').text();
    expect(decrementCount).toBe('0');
  });

  // Tag <Table /> check

  it('dive to table tag', () => {
    const wrapper = shallow(<Counter />);
    // find the <Table />
    let tableTag = wrapper
      .find(Table)
      .dive()
      .find('center')
      .find('[data-test="component-table"]');
    // console.log(tableTag.debug());
    expect(tableTag.length).toBe(1);
  });

  // SnapShot Checking
  it('Snapshot- 1', () => {
    // for specific component snapshot`
    let wrapper = shallow(<Table />);
    // Autometic snapshot created
    expect(wrapper).toMatchSnapshot();
  });
  //  if you want to take another snapshot
  it('Snapshot- 2', () => {
    // for specific component snapshot`
    let wrapper = shallow(<Table />);
    // Autometic snapshot created
    expect(wrapper).toMatchSnapshot();
  });

  //  if you want to take another snapshot
  it('Snapshot- 3', () => {
    // for specific component snapshot`
    let wrapper = shallow(<Table />);
    // Autometic snapshot created
    expect(wrapper).toMatchSnapshot();
  });
});
