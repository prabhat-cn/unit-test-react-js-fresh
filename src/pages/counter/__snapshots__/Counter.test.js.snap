// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Snapshot- 1 1`] = `
<center>
  <div
    className="tableCls custom-border"
    data-test="component-table"
  >
    <table>
      <thead>
        <tr>
          <th>
            #Sl.No
          </th>
          <th>
            Employee Name
          </th>
          <th>
            Company Name
          </th>
          <th>
            Address
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            Rahul Kumar
          </td>
          <td>
            TCS
          </td>
          <td>
            America
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</center>
`;

exports[`Counter Component Snapshot- 2 1`] = `
<center>
  <div
    className="tableCls custom-border"
    data-test="component-table"
  >
    <table>
      <thead>
        <tr>
          <th>
            #Sl.No
          </th>
          <th>
            Employee Name
          </th>
          <th>
            Company Name
          </th>
          <th>
            Address
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            Rahul Kumar
          </td>
          <td>
            TCS
          </td>
          <td>
            America
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</center>
`;

exports[`Counter Component Snapshot- 3 1`] = `
<center>
  <div
    className="tableCls custom-border"
    data-test="component-table"
  >
    <table>
      <thead>
        <tr>
          <th>
            #Sl.No
          </th>
          <th>
            Employee Name
          </th>
          <th>
            Company Name
          </th>
          <th>
            Address
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            Rahul Kumar
          </td>
          <td>
            TCS
          </td>
          <td>
            America
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</center>
`;
